wfl/generate/md/__init__.py

```diff
@@ -206,9 +218,13 @@ def process_step(interval):
             at.info['MD_time_fs'] = cur_step * dt
             traj.append(at_copy_save_results(at, results_prefix=results_prefix))
 
-        # save config_type
-        for at in traj:
-            config_type_append(at, 'MD')
+        if traj_subsampling_fun is not None:
```

Comment:
Should this also be done on the fly, like the name of the PR suggests?

Reply:
I thought having it at the end would be more general - e.g. for CUR you'd need the whole trajectory, no?

Reply:
More general, but problematic if the trajectory is very long.  I think it'd be easy enough to call one function that returns True/False mid trajectory, and another that returns the entire selected set at the end.  

We could wait until we have a specific use case, but it's so simple, I don't think it's an issue to add both function calls now. Do we want a simple function, or an object like the failure test?

Reply:
I generalized it to `traj_select_during_func` and `traj_select_after_func`. See my upcoming commit.

Reply:
I think that's a very good solution, thanks